workspace-loader/src/json-rpc/util.ts

```diff
@@ -22,8 +22,8 @@ export class Deffered<T> implements IDeffered<T> {
     private rejectF;
     constructor() {
         this.promise = new Promise((resolve, reject) => {
-            this.resolve = resolve;
-            this.reject = reject;
+            this.resolveF = resolve;
```

Comment:
I don't get why would you change this to `resolveF`, mind giving a exact reason why this definition(s) needs to be renamed as such?

Reply:
@sr229 Do you think this will work well,  `(new Deffered<any>(promise, reject)).resolve(undefined);` ?

Reply:
Please read the whole code again. The behavior is obvious. `(new Deffered<any>(promise, reject)).resolve(undefined);` will show `TypeError: undefined is not a function` as `resolveF` has never been initialized.

Fortunately or unfortunately, this error will not shown on the current master branch.
This is the reason why I named `Fix a *potentially* bug`

BTW, do you know `Deffered` design pattern?

Reply:
@monaka a bit confused but yes I got the idea. This is meant to resolve #12063 - so I believe you tested this patch?

Reply:
The first step of resolving #12063. It needs some more patches for fixing that completely. The aim of this patch is to fix a obvious unsuitable code. An good application is from clean code, right?

I reviewed all code in Workspace-loader, run tests with Maven, traced on my Chrome. I caught no side-effect.